### PR TITLE
Improve testing documentation

### LINKED ISSUE #88: RSpec view spec, missing policy authorization context
### Tell us about your environment

**Ruby Version:** ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-darwin18]

**Framework Version (Rails, whatever):** Rails 6.0.0

**Action Policy Version:** 0.3.2

### What did you do?

I'm testing some views of my app that use `allowed_to?` helper.

### What did you expect to happen?

Having my test working normally. Or to know what could I change in my test to avoid this.

### What actually happened?

```
Failures:

  1) admin/users/edit.html.slim displays user#edit correctly 
     Failure/Error: - if allowed_to?(:edit?, @user)
     
     ActionView::Template::Error:
       Missing policy authorization context: user
     # ./app/views/admin/users/_form.html.slim:19:in `block in _app_views_admin_users__form_html_slim__2614983762995680549_70232191255660'
     # ./app/views/admin/users/_form.html.slim:9:in `_app_views_admin_users__form_html_slim__2614983762995680549_70232191255660'
     # ./app/views/admin/users/edit.html.slim:1:in `_app_views_admin_users_edit_html_slim__69430724083846117_70232190867720'
     # ./spec/views/admin/users/edit.html.slim_spec.rb:13:in `block (3 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # ActionPolicy::AuthorizationContextMissing:
     #   Missing policy authorization context: user
     #   ./app/views/admin/users/_form.html.slim:19:in `block in _app_views_admin_users__form_html_slim__2614983762995680549_70232191255660'
```

### PR DESCRIPTION
**Pull request by nicolas-brousse**
### What is the purpose of this pull request?

Update the testing documentation.

### What changes did you make? (overview)

I added section about `Testing params scopes` and `Testing views`.

### Is there anything you'd like reviewers to focus on?

I not sure about the section `Testing views`.

PR checklist:

- [ ] Tests included
- [x] Documentation updated
- [ ] Changelog entry added

Closes #88, #105

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -331,3 +360,30 @@ expect { subject }.to have_authorized_scope(:scope)
...
+
+## Testing views
+
+When you test views that call policies methods as `allowed_to?`, your may have `Missing policy authorization context: user` error.
+You may need to stub `current_user` to resolve the issue.
+
+Consider an RSpec example:
+
+```ruby
+RSpec.describe "users/index.html.slim", type: :view do
```
Comment (palkan) [inline docs/testing.md]: Let's stick to the existing RSpec style we already have in this doc:

```suggestion
describe "users/index.html.slim" do
```

File: docs/testing.md [hunk 2]
```diff
@@ -156,6 +156,35 @@ describe PostPolicy do
 end
 ```
 
+### Testing params scopes
```
Comment (palkan) [inline docs/testing.md]: Let's make this a subsection of `Testing scopes`, smth like:

```suggestion
#### Action Controller params example
```

And add a similar heading for Active Record (`Active Record relation example`).

**Comment by nicolas-brousse** [comment]
@palkan thanks for the review. I did the changes 🙂

File: docs/testing.md [hunk 3]
```diff
@@ -110,7 +110,7 @@ OR
 
 **NOTE:** the DSL is included only to example with the tag `type: :policy` or in the `spec/policies` folder. If you want to add this DSL to other examples, add `include ActionPolicy::RSpec::PolicyExampleGroup`.
 
-### Testing scopes
+#### Active Record relation example
```
Comment (palkan) [inline docs/testing.md]: Let's leave the previous header

```suggestion
### Testing scopes

#### Active Record relation example
```

Comment (nicolas-brousse) [reply docs/testing.md]: Sorry, I misunderstood.

File: docs/testing.md [hunk 4]
```diff
@@ -156,6 +158,35 @@ describe PostPolicy do
...
+    subject { policy.apply_scope(target, type: :action_controller_params).to_h }
+
+    context "as user" do
+      it { is_expected.to eq({name: "a"}) }
+    end
+
+    context "as manager" do
+      before { user.update!(role: :manager) }
+
+      it { is_expected.to eq({name: "a", password: "b"})  }
```
Comment (palkan) [inline docs/testing.md]:
```suggestion
      it { is_expected.to eq({name: "a", password: "b"}) }
```

**Pull request merged**